lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -351,6 +351,23 @@ bool SPIRVRegularizeLLVMBase::regularize() {
             II.setMetadata(MDName, nullptr);
           }
         }
+        if (auto *ASCast = dyn_cast<AddrSpaceCastInst>(&II)) {
```

Comment:
It would be nice to have a comment giving a one-line summary of what is happening inside this if-block.  Something like (mostly taken from your description):
"Add an additional bitcast in case address space cast also changes pointer element type".

Reply:
Sure, done.

Reply:
Thanks!